Source/JamfProClient.swift

```diff
@@ -102,10 +111,11 @@ struct JamfProClient {
                     let major = Int(versionParts[0]),
                     let minor = Int(versionParts[1]),
                     let patch = Int(versionParts[2]) {
-                    result = (major: major, minor: minor, patch: patch)
+                    version = Version(major: major, minor: minor, patch: patch)
                 }
             }
-            completionBlock(result)
+            let connectionOk = statusCode == 200 || statusCode == 401
```

Comment:
Doesn't 401 mean unauthorized? Why are we saying that connection is ok?

Reply:
What about just passing the status code into the completion block directly like the `getOrganizationName` method does? Then we can decide how to handle it when we call the method in the `UploadViewController`.

Reply:
Yes, it means unauthorized. Because we just get content of main page `https://something.jamfcloud.com` there is a login page and for some reason, JSS returns it together with code 401.

I tried to pass statusCode as you suggested but I think that this logic `connectionOk = statusCode == 200 || statusCode == 401` make more sense inside this method. When I moved it outside it was harder to understand. I changed the order of arguments thought to match `getOrganizationName`.

Reply:
Gotcha, that works for me. Thanks for the clarity.